File: src/main/java/com/jozufozu/exnihiloomnia/common/lib/LibItems.java
package com.jozufozu.exnihiloomnia.common.lib;

import com.jozufozu.exnihiloomnia.ExNihilo;
import net.minecraft.util.ResourceLocation;

public class LibItems
{
    /**
     * Material prefixes
     */
    public static final String WOOD = "_wood";
    public static final String STONE = "_stone";
    public static final String IRON = "_iron";
    public static final String GOLD = "_gold";
    public static final String DIAMOND = "_diamond";
    
    /**
     * Ex Nihilo materials
     */
    public static final String SILK = "_silk";
    public static final String BONE = "_bone";
    
    /**
     * Tools
     */
    public static final String MESH = "mesh";
    public static final String CROOK = "crook";
    public static final String HAMMER = "hammer";
    
    /**
     * Mesh
     */
    public static final ResourceLocation WOODEN_MESH = get(MESH, WOOD);
    public static final ResourceLocation SILK_MESH = get(MESH, SILK);
    public static final ResourceLocation GOLD_MESH = get(MESH, GOLD);
    public static final ResourceLocation DIAMOND_MESH = get(MESH, DIAMOND);
    
    public static final ResourceLocation WOODEN_HAMMER = get(HAMMER, WOOD);
    public static final ResourceLocation STONE_HAMMER = get(HAMMER, STONE);
    public static final ResourceLocation IRON_HAMMER = get(HAMMER, IRON);
    public static final ResourceLocation GOLD_HAMMER = get(HAMMER, GOLD);
    public static final ResourceLocation DIAMOND_HAMMER = get(HAMMER, DIAMOND);
    
    public static final ResourceLocation TREE_SEED = get("tree_seed");
    public static final ResourceLocation SMALL_STONE = get("small_stone");
    
    private static ResourceLocation get(String... name)
    {
        StringBuilder builder = new StringBuilder();
        for (String s : name)
        {
            builder.append(s);
        }
        
        return new ResourceLocation(ExNihilo.MODID, builder.toString());
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/lib/LibItems.kt
package com.jozufozu.exnihiloomnia.common.lib

import com.jozufozu.exnihiloomnia.ExNihilo
import net.minecraft.util.ResourceLocation

object LibItems {
    /**
     * Material prefixes
     */
    const val WOOD = "wooden"
    const val STONE = "stone"
    const val IRON = "iron"
    const val GOLD = "gold"
    const val DIAMOND = "diamond"

    const val PORCELAIN = "porcelain"

    /**
     * Ex Nihilo materials
     */
    const val SILK = "silk"
    const val BONE = "bone"

    /**
     * Tools
     */
    const val MESH = "mesh"
    const val HAMMER = "hammer"
    const val CROOK = "crook"

    /**
     * Mesh
     */
    val WOODEN_MESH = get(WOOD, MESH)
    val SILK_MESH = get(SILK, MESH)
    val GOLD_MESH = get(GOLD, MESH)
    val DIAMOND_MESH = get(DIAMOND, MESH)

    val WOODEN_CROOK = get(WOOD, CROOK)
    val BONE_CROOK = get(BONE, CROOK)

    val WOODEN_HAMMER = get(WOOD, HAMMER)
    val STONE_HAMMER = get(STONE, HAMMER)
    val IRON_HAMMER = get(IRON, HAMMER)
    val GOLD_HAMMER = get(GOLD, HAMMER)
    val DIAMOND_HAMMER = get(DIAMOND, HAMMER)

    val TREE_SEED = get("tree_seed")
    val SMALL_STONE = get("small_stone")

    val ASTROLABE = get("astrolabe")
    val ASH = get("ash")
    val SILKWORM = get("silkworm")

    val PORCELAIN_CLAY = get(PORCELAIN, "clay")

    val PORCELAIN_BUCKET = get(PORCELAIN, "bucket")

    private fun get(vararg name: String): ResourceLocation {
        return ResourceLocation(ExNihilo.MODID, name.joinToString("_"))
    }
}